HeroVault/HeroVault/CoreData/CoreDataHandler.swift

```diff
@@ -169,14 +176,23 @@ class CoreDataHandler: CoreDataHandlerType {
         }
     }
     
-    func deleteUsers() {
-        let fetchRequest: NSFetchRequest<NSFetchRequestResult> = User.fetchRequest()
-        let batchDeleteRequest = NSBatchDeleteRequest(fetchRequest: fetchRequest)
-        _ = try? context.execute(batchDeleteRequest)
-    }
-    
     // MARK: - Core Data Helper Functions
     
+    func hasObjectBeenFavourited(_ object: MarvelData, entityType: EntityType) -> Bool {
+        
+        let fetchRequest = NSFetchRequest<NSManagedObject>(entityName: entityType.rawValue)
+        fetchRequest.predicate = NSPredicate(format: "id == %d", object.id)
+        
+        do {
+            let results = try context.fetch(fetchRequest)
+            let marvelData = results.first as? MarvelData
+            return marvelData?.hasBeenfavourited ?? false
+        } catch {
+            print("Error fetching character: \(error.localizedDescription)")
+            return false
+        }
+    }
+    
```

Comment:
// MARK: - Core Data Helper Functions
 this mark should be here by the looks of it and can be removed from where it is